### PR TITLE
feat: Resolve recursive references

### PR DESCRIPTION
**Pull request by rtfpessoa**
One feature I forgot to mention before, that was fairly simple to implement was recursive references.

### CODE REVIEW

File: src/generate.ts [hunk 1]
```diff
@@ -180,14 +180,18 @@ function supportDeepObjects(params: OpenAPIV3.ParameterObject[]) {
 export default function generateApi(spec: OpenAPIV3.Document) {
   const aliases: ts.TypeAliasDeclaration[] = [];
 
-  function resolve<T>(obj: T | OpenAPIV3.ReferenceObject) {
-    if (!isReference(obj)) return obj;
-    const ref = obj.$ref;
-    if (!ref.startsWith("#/")) {
-      throw new Error(`External refs are not supported: ${ref}`);
+  function resolve<T>(obj: T | OpenAPIV3.ReferenceObject): T {
+    function resolveAux<T>(obj: T | OpenAPIV3.ReferenceObject): T {
```
Comment (fgnass) [inline src/generate.ts:184]: Why do we need this inner function? Couldn't `resolve` call itself in line 191?

**Comment by fgnass** [comment]
Thank you! Can you add a test case for recursive references?

File: src/generate.ts (hunk 1 above)
Comment (rtfpessoa) [reply src/generate.ts:184]: You are completely right.

**Comment by rtfpessoa** [comment]
Forget about this, it is already supported.
I mistakenly tough this would be needed but it is not, since I need something similar for the validation, but while doing the test I might found a better way of doing that.
Sorry for the confusion.

**Pull request closed without merging**